x3270: init at 3.4ga9

musteresel opened the pull request:
I added a nix expression to build the x3270 IBM 3270 terminal emulator, to be found there:

http://x3270.bgp.nu/index.html

I packaged the current stable version (3.4ga9 stable (23. October 2015), according to their downloads page). Though I pull the source code from the git repository instead of the provided tar balls. A small patch to fix the interpreter of an utility bash script was needed.

Actually this package now provides multiple binaries. One could add support to only build a subset of these binaries (and thus removing tcl, x dependencies). Also there are no man pages, because the respective install target fails and I don't have the time to investigate further.

As for testing I'm successfully running the applications on my laptop (NixOS on x86_64), using them to participate in the IBM Master the Mainframe contest. No further testing than this was done, though.

Please let me know if there's something that needs to be changed (this is my first to be contributed package). This is a second pull request, the first #11031 targeted the release branch and I couldn't get it to target the master branch.

musteresel committed:
x3270: init at 3.4ga9

x3270 is an IBM 3270 terminal emulator for the X Window System. It comes
with some other - related - tools, such as a ncurses version or a
version intended for use with scripts. These tools are currently
unconditionally built, this is a possible point of improvement. Also the
man pages unfortunately don't install with this nix expression.

Version 3.4ga9 is the current (23. Oct 2015) stable version of this
software.

pSub commented on pkgs/applications/misc/x3270/default.nix:
```diff
@@ -0,0 +1,27 @@
+
+  patches = [ ./repair-interpreter.patch ];
+  
+  meta = {
```
Could you add `meta.license`?

pSub commented on pkgs/applications/misc/x3270/default.nix:
```diff
@@ -0,0 +1,27 @@
+
+stdenv.mkDerivation {
+  name = "x3270-3.4ga9";
+  src = fetchgit {
```
Are there any tarball releases? If so, please use `fetchurl`.

pSub commented on pkgs/applications/misc/x3270/default.nix:
```diff
@@ -0,0 +1,27 @@
+  
+  meta = {
+    homepage = http://x3270.bgp.nu/index.html;
+    description = "x3270 is an IBM 3270 terminal emulator for the X Window System.";
```
`description` should not contain the package name and should not end with a period. See the nixpkgs manual for details.

musteresel committed:
Use provided tarball release, though untested

musteresel commented:
@pSub I added the licence as well as corrected the description. I amended the "main commit" with these changes. The homepage of the tool does provide source tarballs of the current stable release. I thought that using the tag from the repository would be the more stable approach, but I added another commit using fetchurl and this tarball instead. Though I cannot test it locally on my machine atm.

rickynils commented:
@musteresel It seems your latest commit (2d79062) didn't work: https://travis-ci.org/NixOS/nixpkgs/builds/91654817. Also, the commit description should ideally include the package it changes.

rasendubi commented:
(triage) @musteresel do you expect to work on this futher?

grahamc commented:
Closing. Can re-open if you would like :)
